core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -96,7 +105,21 @@ oppia.factory('explorationData', [
               $log.info(response);
               draftChangeListId = response.draft_change_list_id;
               explorationData.data = response;
-
+              var draft = LocalStorageService.getExplorationDraft(
+                explorationId);
+              if (draft &&
+                draft.isDraftValid(draftChangeListId)) {
+                var changeList = draft.getDraftChanges();
+                explorationData.autosaveChangeList(changeList, function() {
+                  // A reload is needed so that the changelist just saved is
+                  // loaded as opposed to the exploration returned by this
+                  // response.
+                  $window.location.reload();
+                });
+              } else if (draft &&
```

Comment:
Better to do:

    if (draft) {
      if (draft.isDraftValid(draftChangeListId)) {
        ...
      } else {
        ...
      }
    }

Reply:
Done.